fsharp-backend/src/LibExecution/StdLib/LibString.fs

```diff
@@ -419,15 +420,13 @@ let fns : List<BuiltInFn> =
       fn =
         (function
         | _, [ DStr s ] ->
-            let append (reversedString : char list) (ch : char) : char list =
-              ch :: reversedString
+            let reverseString str =
+                StringInfo.ParseCombiningCharacters(str)
```

Comment:
I'd suggest using Prelude.String.toEgcSeq to make a sequence of EGCs (which are what dark uses for characters) and then reversing and then using String.concat.